sql-plugin/src/main/330+/scala/com/nvidia/spark/rapids/shims/v2/Spark33XShims.scala

```diff
@@ -143,4 +143,16 @@ trait Spark33XShims extends Spark33XFileOptionsShims {
             conf.maxReadBatchSizeBytes)
       })
   ).map(r => (r.getClassFor.asSubclass(classOf[Scan]), r)).toMap
+
+  override def tagHiddenMetadataExpressions(meta: SparkPlanMeta[FileSourceScanExec]): Unit = {
+    val metadataColumns = meta.wrapped.expressions.filter { expr =>
+      expr match {
+        case MetadataAttribute(expr) => true
+        case _ => false
+      }
+    }
+    if (metadataColumns.nonEmpty) {
```

Comment:
Nit: We don't actually care which columns are the metadata columns, just if there is at least one.
```suggestion
    val hasMetadataColumns = meta.wrapped.expressions.exists { expr =>
      expr.isInstanceOf[MetadataAttribute]
    }
    if (hasMetadataColumns) {
```

Reply:
I will make the `exists` change but we need to use `MetadataAttribute.unapply` here rather than `.isInstanceOf[MetadataAttribute]` since the expressions themselves are `AttributeReference`